src/Features/VisualBasic/Portable/CodeRefactorings/AddNamedArguments/VisualBasicAddNamedArgumentsCodeRefactoringProvider.vb

```diff
@@ -0,0 +1,133 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Composition
+Imports System.Collections.Immutable
+Imports Microsoft.CodeAnalysis.CodeRefactorings
+Imports Microsoft.CodeAnalysis.CodeRefactorings.AddNamedArguments
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+Imports Microsoft.CodeAnalysis.VisualBasic
+Imports Microsoft.CodeAnalysis.VisualBasic.SyntaxFactory
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.CodeRefactorings.AddNamedArguments
+    <ExtensionOrder(After := PredefinedCodeRefactoringProviderNames.IntroduceVariable)>
+    <ExportCodeRefactoringProvider(LanguageNames.VisualBasic, Name:=NameOf(VisualBasicAddNamedArgumentsCodeRefactoringProvider)), [Shared]>
+    Friend Class VisualBasicAddNamedArgumentsCodeRefactoringProvider
+        Inherits AbstractAddNamedArgumentsCodeRefactoringProvider(Of ArgumentSyntax)
+
+        Protected Overrides Function TryGetOrSynthesizeNamedArguments(arguments As SeparatedSyntaxList(Of ArgumentSyntax), parameters As ImmutableArray(Of IParameterSymbol), ByRef namedArguments As ArgumentSyntax(), ByRef hasLiteral As Boolean) As Boolean
+            ' An argument list with omitted arguments may or may not be already fixed
+            Dim alreadyFixed = True
+            Dim result = ArrayBuilder(Of ArgumentSyntax).GetInstance(arguments.Count)
+            For index = 0 To arguments.Count - 1
+                Dim argument = arguments(index)
+                Dim parameter = parameters(index)
+                If argument.IsNamed Then
+                    result.Add(argument)
+                ElseIf argument.IsOmitted Then
+                    Continue For
+                ElseIf parameter.IsParams Then
+                    ' Named arguments cannot match a ParamArray parameter
+                    result.Free()
```

Comment:
just wrap in a try/finally and put the 'free' in there.  Then you don't have to worry about cleaning up through every codepath.